lib/db_connection/connection.ex

```diff
@@ -380,7 +386,7 @@ defmodule DBConnection.Connection do
     do_handle_info(msg, s)
   end
   def handle_info(:timeout, %{idle_hibernate: idle_hibernate} = s) do
-    maybe_hibernate(idle_hibernate, {:noreply, s})
+    maybe_hibernate(idle_hibernate, {:noreply, %{s | hibernate_timer: nil}})
```

Comment:
We should never reach this `handle_info/2` clause. I think we can delete this clause.